File: NewBlackbox/app/src/main/java/com/github/nukc/stateview/StateView.java
package com.github.nukc.stateview;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

import androidx.annotation.Nullable;

public class StateView extends FrameLayout {

    public StateView(Context context) {
        super(context);
    }

    public StateView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public StateView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void showLoading() {
        setVisibility(VISIBLE);
    }

    public void showContent() {
        setVisibility(VISIBLE);
    }

    public void showEmpty() {
        setVisibility(VISIBLE);
    }
}

File: NewBlackbox/app/src/main/java/com/github/nukc/stateview/StateView.kt
package com.github.nukc.stateview

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout

class StateView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : FrameLayout(context, attrs, defStyleAttr) {

    fun showLoading() {
        visibility = VISIBLE
    }

    fun showContent() {
        visibility = VISIBLE
    }

    fun showEmpty() {
        visibility = VISIBLE
    }
}
